cmd/sops/common/common.go

```diff
@@ -447,3 +454,34 @@ func PrettyPrintDiffs(diffs []Diff) {
 		}
 	}
 }
+
+type ShamirThresholdDiff struct {
+	Old int
+	New int
+}
+
+// DiffShamirThreshold Use config (limited to number of key groups) if there are two or more groups
+func DiffShamirThreshold(metadata sops.Metadata, conf *config.Config) ShamirThresholdDiff {
+	var diff = ShamirThresholdDiff{}
+	diff.Old = metadata.ShamirThreshold
+	if conf.ShamirThreshold > 0 && len(conf.KeyGroups) > 1 {
+		diff.New = min(conf.ShamirThreshold, len(conf.KeyGroups))
+	} else if len(conf.KeyGroups) > 1 {
+		diff.New = len(conf.KeyGroups)
+	}
+
+	return diff
+}
```

Comment:
I don't think the logic to compute the new shamir theshold should be part of a diff function. I think it should be a separate function, that can be used in the diff function (though I would avoid that and simply pass old and new value on after the new value has been computed).